system-addon/lib/TopSitesFeed.jsm

```diff
@@ -146,12 +146,12 @@ this.TopSitesFeed = class TopSitesFeed {
   pin(action) {
     const {site, index} = action.data;
     NewTabUtils.pinnedLinks.pin(site, index);
-    this._broadcastPinnedSitesUpdated();
+    this.refresh();
   }
   unpin(action) {
     const {site} = action.data;
     NewTabUtils.pinnedLinks.unpin(site);
-    this._broadcastPinnedSitesUpdated();
+    this.refresh();
```

Comment:
@rlr has said that we purposely avoid refresh on pins per @k88hudson's suggestion. Now with deduping, the new pinned site might cause some frecent or default sites to disappear when correctly refreshed.

I'm not sure if this means we should just call `refresh` or update `refresh -> getLinksWithDefaults` to conditionally fetch new links from `NewTabUtils`?

Reply:
What's the potential deduping? The user adds/edits in a frecent or default site? The top sites list could change after 15 minutes anyway from just regular browsing, so it's not wrong for things to change.

Reply:
Yes, it's a matter of do we want to do it ourselves and update it instantly or let the 15 minutes pass.
This is more obvious when you change the URL but the dedupe key (hostname) is the same. If we call refresh ourselves we won't get 2 websites after closing the edit form just 1.
I'll remove the `refresh` calls and go forward with the solution. I can easily add them back in later.